enhancements/microshift/microshift-csi-snapshot-integration.md

```diff
@@ -0,0 +1,447 @@
+for MicroShift’s control-plane elements.
+
+The CSI Snapshotter configuration is managed via the VolumeSnapshotClass API. This API serves a similar purpose as 
+StorageClasses and allows admins to specify dynamic snapshotting parameters at runtime.  MicroShift will deploy a 
+default VolumeSnapshotClass on startup. This instance will reference the default StorageClass that is already 
+deployed by MicroShift to enable snapshotting out of the box.
+
+The MicroShift deployment model utilizes rpm-ostree layers. Following the existing deployment pattern, 
+VolumeSnapshotClass manifests can be packaged and deployed onto target devices.
+(See [kubernetes-for-device-edge. md#workflow-description.md](./kubernetes-for-device-edge.md#workflow-description)).
+Application deployers can predefine VolumeSnapshotClass manifests.  These can be written the manifests to 
+/etc/microshift/manifests. This pattern is congruent with how a device owner would install custom StorageClasses. It 
+is recommended that new custom StorageClasses be packaged with VolumeSnapshotClasses that reference them.
+
+#### Upgrading
+
+CSI components are upgraded by MicroShift-internal automation.  These scripts pull the CSI image digests and 
+related manifests from the OpenShift LVMS Operator Bundle.  This helps to keep MicroShift's version of 
+CSI compatible with the default storage provisioner (LVMS). MicroShift deploys CSI `v1`, with micro-versions being 
+guaranteed backwards compatible within each major version.  Therefore, micro-version upgrades will not require user 
+intervention to protect snapshots from deletion or orphaning.
+
+Upgrading the CSI containers will introduce a small interrupt window to provisioning operations.  Therefore, 
+to mitigate any risk or orphaning resources, it is recommended that snapshotting be allowed to complete before 
+restarting MicroShift and initiating the upgrade.
+
+#### Configuring
+
+Cluster admins will use the VolumeSnapshotClass API to set dynamic configurations for snapshot creation. 
+Driver-specific configuration is available through the ‘parameters’ sub-field, which is a string:string map and is 
+defined by the particular storage provider.
+
+#### Deploying Applications
+
+### API Extensions
+
+The APIs to be added are defined under the `snapshot.storage.k8s.io/v1` API group. They are:
+
+1. VolumeSnapshot
+2. VolumeSnapshotClass
+3. VolumeSnapshotContent
+
+For a detailed description of these APIs, see 
+[OpenShift documentation](https://docs.openshift.com/container-platform/4.13/storage/container_storage_interface/persistent-storage-csi-snapshots.html#volume-snapshot-crds)
+
+### Risks and Mitigations
+
+- Increased overhead.  The CSI snapshot controller and sidecar increase the on-disk footprint by a
+total of 360MB.  At idle, the components use a negligible amount of CPU (>1% CPU on a 2 core machine)
+and roughly 18Mb.
+
+### Drawbacks
+
+## Design Details
```

Comment:
It looks like no new components need to be vendored into the MicroShift binary because everything will run in a pod. Is that correct?